### PR TITLE
feat: Add components & embeds

### LINKED ISSUE #7: Use embeds and message components for nicer UX
As mentioned in #3, should use embeds for configs, and I am also adding message components for editing them.

I tried to do this yesterday and it was kinda messy, after doing #6 I've gotten used to the codebase a bit so I'll probably try to tackle it again after it's merged (so I have a sane starting point)

### PR DESCRIPTION
**Pull request by alyti**
Also formats code & imports humantime for nicer ttl.

Closes #7

### CODE REVIEW

**Comment by alyti** [comment]
@RaphaelDarley Alright I am finished with this now I think, there's a few things that could still be done like having a hashmap for user to query/vars for Big Query editor (I added a TODO for that), but that's also more state to maintain so I am holding off on that for now.

Compared to what I showed you yesterday, I added the missing handlers for ttl & timeout, I also added json variables to Big Query.

**Comment by alyti** [comment]
I haven't really touched much on the responses from commands, so they could still be updated to use the new `success_ephemeral_interaction`/`failure_ephemeral_interaction`/`success_user_interaction`, which are interaction-source independent, as opposed to the current *command-only* ones. `Query`/`Export` commands could also switch to using the methods on Conn type, see session component & modal handlers for reference.

File: src/components/configurable_session.rs [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+                .get_mut(&channel.0)
+                .expect("DB disappeared between now above check")
+                .clone();
+            match id {
+                "format" => db.json = values[0] == "json",
+                "prettify" => db.pretty = values[0] == "true",
+                "require_query" => db.require_query = values[0] == "true",
+                _ => unreachable!(),
+            }
+            DBCONNS.lock().await.insert(channel.0, db);
```
Comment (RaphaelDarley) [inline src/components/configurable_session.rs:111]: if you did get_mut you shouldn't need to insert as well. Or if you are going to clone then you don't need get_mut

File: src/components/configurable_session.rs [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+    channel: &ChannelId,
+    id: &str,
+    values: &[String],
+) -> Result<()> {
+    let db_exists = DBCONNS.lock().await.get(&channel.0).is_some();
+
+    match (id, db_exists) {
+        ("format", true) | ("prettify", true) | ("require_query", true) => {
+            debug!("Updating config");
+            let mut db = DBCONNS
```
Comment (RaphaelDarley) [inline src/components/configurable_session.rs:99]: db should be of type Conn so the name is inconsistent

**Review by RaphaelDarley — APPROVED** [review]
Looks, big improvement to ux

**Pull request merged**